From: Ursula Braun <ubraun@linux.vnet.ibm.com>
Subject: s390/qeth: don't convert return code twice
Patch-mainline: v4.11-rc1
Git-commit: 4b764d1de395090662fd0291968d5ded523f07e4
References: bsc#1025380, FATE#321529, LTC#146544

Summary:     qeth: optimize IP handling in rx_mode callback
Description: In layer3 mode of the qeth driver, multicast IP addresses
             from struct net_device and other type of IP addresses
             from other sources require mapping to the OSA-card.
             The IP address mapping logic is simplified, and the
             implementation of ndo_set_rx_mode callback and ip notifier events
             is changed. Addresses are stored in private hashtables instead of
             lists now. It allows hardware registration/removal for new/deleted
             multicast addresses only.

Upstream-Description:

             s390/qeth: don't convert return code twice

             qeth_l2_send_groupmac() already translates the return code, so
             calling qeth_setdel_makerc() a second time only produces garbage.

             Signed-off-by: Julian Wiedmann <jwi@linux.vnet.ibm.com>
             Reviewed-by: Thomas Richter <tmricht@linux.vnet.ibm.com>
             Reviewed-by: Ursula Braun <ubraun@linux.vnet.ibm.com>
             Signed-off-by: David S. Miller <davem@davemloft.net>


Signed-off-by: Ursula Braun <ubraun@linux.vnet.ibm.com>
Acked-by: John Jolly <jjolly@suse.com>
---
 drivers/s390/net/qeth_l2_main.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

--- a/drivers/s390/net/qeth_l2_main.c
+++ b/drivers/s390/net/qeth_l2_main.c
@@ -210,8 +210,7 @@ static int qeth_l2_write_mac(struct qeth
 				qeth_l2_send_setdelmac(card, mac->mac_addr,
 						IPA_CMD_SETVMAC));
 	} else {
-		rc = qeth_setdel_makerc(card,
-				qeth_l2_send_setgroupmac(card, mac->mac_addr));
+		rc = qeth_l2_send_setgroupmac(card, mac->mac_addr);
 	}
 	return rc;
 }
